openedx_events/learning/data.py

```diff
@@ -296,3 +296,17 @@ class ProgramCertificateData:
     status = attr.ib(type=str)
     url = attr.ib(type=str)
     certificate_available_date = attr.ib(type=datetime, default=None)
+
+
+@attr.s(frozen=True)
+class CourseStaffData:
+    """
+    Attributes defined for the Open edX Authentication function to Add Users to a role.
+
+    Arguments:
+        course_key (Course ID): identifier of the course where the user will have staff role assigned/removed
+        user (UserData): User who will have the "Course Staff" role assigned/removed
+    """
+
+    course_key = attr.ib(type=str)
+    user = attr.ib(type=UserData)
```

Comment:
which other fields do you think would be useful?

Reply:
I added some other fields based on the current parameters that CourseAccessRole currently takes: https://github.com/openedx/edx-platform/blob/master/common/djangoapps/student/roles.py#L448

Please note that this is subject to change in the future due to the concurrent roles/permissions project that is being done (see the "IMPORTANT:" comment in the event data for more details).

Reply:
That's better. Thank you!